File: Proxy/Java/CuentaProxy.java

public class CuentaProxy implements Icuenta {
    //private CuentaBanco cuentaReal;
    private Icuenta cuentaReal;

    public CuentaProxy(Icuenta cuentaReal){
        this.cuentaReal = cuentaReal;
    }
     
    @Override
    public Cuenta retirarDinero(Cuenta cuenta, double monto){
        if (cuentaReal == null){ 
            cuentaReal = new CuentaBanco();
             return cuentaReal.retirarDinero(cuenta, monto);
        } else{
            return cuentaReal.retirarDinero(cuenta, monto);
        }
    }

    public Cuenta depositarDinero(Cuenta cuenta, double monto){
        if (cuentaReal == null){
            cuentaReal = new CuentaBanco();
            return cuentaReal.depositarDinero(cuenta, monto);
        }else{
            return cuentaReal.depositarDinero(cuenta, monto);
        }
    }

    public void mostrarSaldo(Cuenta cuenta){
        if(cuentaReal == null){
            cuentaReal = new CuentaBanco();
            cuentaReal.mostrarSaldo(cuenta);
        }else{
            cuentaReal.mostrarSaldo(cuenta);
        }
    }

}

File: Proxy/Python/CuentaProxy.py
from Icuenta import *
from CuentaBanco import *

class CuentaProxy(Icuenta):

    _cuentaReal = None
    
    def __init__(self, cuentaReal):
        print("ME he creado")
        self._cuentaReal = cuentaReal

    def retirarDinero(self, cuenta, monto):
        if(self._cuentaReal == None ):
            print("Entre 1 ")
            self._cuentaReal = CuentaBanco()
            return self._cuentaReal.retirarDinero(cuenta, monto)
        else:
            print("entre dos")
            return self._cuentaReal.retirarDinero(cuenta, monto)

    def depositarDinero(self, cuenta, monto):
        if(self._cuentaReal == None):
            "entre 3"
            self._cuentaReal = CuentaBanco()
            return self._cuentaReal.depositarDinero(cuenta, monto)
        else:
            "entre 4"
            return self._cuentaReal.depositarDinero(cuenta, monto)
    
    def mostrarSaldo(self, cuenta):
        if(self._cuentaReal == None):
            self._cuentaReal = CuentaBanco()
            return self._cuentaReal.mostrarSaldo(cuenta)
        else:
            self._cuentaReal.mostrarSaldo(cuenta)